v2/stack_analyses.yaml

```diff
@@ -0,0 +1,455 @@
+openapi: 3.0.1
+info:
+  title: Stack Analyses API
+  version: "2.0"
+servers:
+  - url: https://f8a-analytics-2445582058137.{environment}.gw.apicast.io/api/v2
+    variables:
+      environment:
+        default: api    # Production server
+        enum:
+          - production # Production server
+          - staging # Staging server
+security:
+  - tokenAuth: []
+paths:
+  /stack-analyses:
+    post:
+      tags:
+      - Scan Services
+      summary: Initiates a scan to gather security and other details about an application
+        stack and related recommendations to improve the stack
+      operationId: f8a_scanner.api_v2.stack-analyses-post
+      parameters:
+      - name: ecosystem
+        in: header
+        description: The ecosystem of the stack analysis being called.
+        required: true
+        schema:
+          type: string
+      - name: origin
+        in: header
+        description: This is required to give an indication as to from where is the
+          request getting generated. If from VScode extension, the value which we
+          need to send is 'vscode'.
+        schema:
+          type: string
+          default: vscode
+          enum:
+            - vscode
+      - name: show_transitive
+        in: header
+        description: This is required to enable or disable the transitive support.
+          We need to pass the value as 'true' if transitive is needed. Default value
+          is 'false'.
+        schema:
+          type: boolean
+          default: true
+      requestBody:
```

Comment:
How requests from free tier / registered users are differentiated ?

Reply:
Once the Auth flow is finalised, most probably we will be sending UUID as part of stack-analyses POST request. UUID will help us in the backend to identify whether the given request is from registered user / un-registered user.

Reply:
UUID creation is something we need to finalize : need to created on the server side. @yzainee ?

Reply:
For vscode, we would still need it to be generated at the client side. Reasons: while we are doing the processing etc, there are chances where the window might be closed or the ext is closed. In all such cases, it will become very difficult to find a way to send back that uuid to the ext to be saved. To avoid such complexities, its best we generate the uuid at ext and for all other integrations, we do it at the server side.

Reply:
@SriKrishnaPaparaju ^^

Reply:
I see that UUID been added as part of the request header.